app/src/main/java/com/bigyoshi/qrhunt/qr/QrLibrary.java

```diff
@@ -45,38 +44,63 @@ public QrLibrary(FirebaseFirestore db, String playerId){
      *
      */
     public void update() {
+        qrCodesList = new ArrayList<>();
         Query qrList =  db.collection("users").document(playerId).collection("qrCodes");
         qrList.get().addOnCompleteListener(task -> {
             if (task.isSuccessful()) {
                 for (QueryDocumentSnapshot doc : task.getResult()) {
                     if (doc.exists()) {
-                        qrCode = doc.toObject(PlayableQrCode.class);
-                        qrCodes.put(doc.getId(), qrCode);
+                        PlayableQrCode qrCode = doc.toObject(PlayableQrCode.class);
+                        qrCodesList.add(qrCode);
                     }
                 }
+                qrCodesList.sort(compareByScore);
             }
         });
+        scoreSorted = 0;
     }
 
-    public HashMap<String, PlayableQrCode> getQrCodes() { return qrCodes; }
+    public ArrayList<PlayableQrCode> getQrCodes() {
+        return qrCodesList;
+    }
 
     /**
      * Sorts all QRs in library from lowest to highest scoring
      *
      */
-    public void sortLowestToHighest(){
+    public ArrayList<PlayableQrCode> sortScoreAscending(){
         /* Integer in HashMap would either be the value of the QRCode
         or just some sort of order we use to rank the QRCodes (ie the values)
          */
+        qrCodesList.sort(compareByScore.reversed());
+        scoreSorted = 1;
+        return qrCodesList;
+
     }
```

Comment:
sort is an in place operation, no need to return it in the future. Not like we're wasting memory anyway, it'll refer to the same object.

Reply:
Does that work when you are accessing the ArrayList from a different file, the current arrayList is private too

Reply:
Its the same object, so yes IINM